src/request.js

```diff
@@ -1,34 +1,77 @@
+// @flow
+
 const EventEmitter = require('events').EventEmitter;
 const TYPES = require('./data-type').typeByName;
 const RequestError = require('./errors').RequestError;
 
+// TODO: Figure out how to type the `rows` parameter here.
+type CompletionCallback = (error: ?Error, rowCount: ?number, rows: any) => void;
+
+type Parameter = {
+  // TODO: `type` must be a valid TDS value type
+  type: any,
+  name: string,
+  value: mixed,
+  output: boolean,
+  length: ?number,
+  precision: ?number,
+  scale: ?number
+};
+
+type ParameterOptions = {
+  output?: boolean,
+  length?: number,
+  precision?: number,
+  scale?: number
+}
+
 module.exports = class Request extends EventEmitter {
-  constructor(sqlTextOrProcedure, callback) {
+  sqlTextOrProcedure: ?string;
+  parameters: Parameter[];
+  parametersByName: { [string]: Parameter };
+  originalParameters: Parameter[];
+  preparing: boolean;
+  canceled: boolean;
+  paused: boolean;
+  userCallback: CompletionCallback;
+  handle: ?any; // TODO: Figure out the type here.
+  error: ?Error;
+  connection: ?any; // TODO: This should be `Connection`, not `any`.
+
+  callback: () => void;
+
+  constructor(sqlTextOrProcedure: ?string, callback: CompletionCallback) {
     super();
 
     this.sqlTextOrProcedure = sqlTextOrProcedure;
     this.parameters = [];
     this.parametersByName = {};
+    this.originalParameters = [];
```

Comment:
I changed the constructor to initialize all properties that are used inside the `Request` class.

This simplifies the type signature of the class, and will help v8 to perform more/better optimizations for this code. 🎉